### ISSUE TITLE
Transfer learning

### ISSUE THREAD

**Issue by gfumagal** [DeepSceneSeg/AdapNet-pp#7]
Thank you for your amazing work.

I am trying to use Adapnet-pp to segment images of a dataset created by myself involving construction sites scenes. Due to the fact that my dataset is quite small, I would like to use the pretrained weights on Freiburg forest dataset and then fine tune the architecture. According to you, which layers should be freezed and which trained? 

Moreover, my dataset has a different number of classes and when I restore the checkpoints it gives me an error, because the shape of last layers depend on the number of classes, which was 6 for Freiburg forest dataset; how should I fix this?

Thank you in advance

**Comment by avalada** [comment]
Edit line 70 in train.py 
from if var.name+' ' in  name_var: to if var.name+' ' in  name_var and ('conv78' not in var.name and 'conv5' not in var.name and not in 'conv911' not in var.name and 'conv912' not in var.name) :
Now, the last two num of classes dependent layers won't be loaded from checkpoint as well as the two auxiliary loss ones.

Our code does not support transfer learning directly. 
An easy hack will be to insert
self.training = False in AdapNet_pp.py prior to line 49
and then again insert
self. training = True in AdapNet_pp.py prior to layer definition which you want to train.
For example, if I want the last two layers (conv layer which reduces to the desired number of classes and the deconv layer which again depends on the number of classes) 
I'll insert it prior to line 127.
